### PR TITLE
Build/init

### PR DESCRIPTION
**Pull request by Thienhuynh95**
## Description

This PR publishes the first version of zonos-typescript-sdk.
- added github workflows
- added build process for the sdk
- added codegen for generating schema and types
- added test to make sure response and error types format are correct

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  "version": "0.0.1",
+  "private": true,
+  "dependencies": {
+    "graphql": "^16.6.0",
+    "graphql-request": "^4.3.0",
+    "graphql-tag": "^2.12.6"
+  },
+  "engines": {
+    "node": "18"
+  },
```
Comment (goodpickle) [inline package.json:12]: What do you think about removing the consuming projects to a particular node version?
```suggestion
```

Comment (evad1n) [inline package.json:12]: Considering our plugins user various lower versions of node (I believe), I wonder if maybe we should expand this range?

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+<!-- Example:
+Error when importing the sdk
+-->
+
+## Description
+<!-- Explain what this Pull Request should do -->
+<!-- Example:
+This PR fixes a bug in our elastic search order query.
+-->
+This PR
```
Comment (evad1n) [inline pull_request_template.md]: What do you think about moving this file inside the `.github` folder, as it organizes better in my opinion.

File: package.json (hunk 1 above)
Comment (Thienhuynh95) [reply package.json:12]: Yah, that's why I remove the engine when preparing for publishing to npm

File: src/gqlRequest.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import {
+  getSdk as getAuthSdk,
+  Sdk as AuthSdk,
+} from 'src/types/generated/graphql.auth.types';
+import {
+  getSdk as getZonosCustomerGraphSdk,
+  Sdk as ZonosCustomerGraphSdk,
+} from 'src/types/generated/graphql.zonos-customer-graph.types';
+
+type Schema = 'zonos-customer-graph' | 'auth';
```
Comment (evad1n) [inline src/gqlRequest.ts]: What is the reason for the `zonos` prefix on one and not the other?

File: codegen/auth.codegen.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import type { TypeScriptDocumentsPluginConfig } from '@graphql-codegen/typescript-operations';
```
Comment (goodpickle) [inline codegen/auth.codegen.ts]: What do you think about using the graph prefix `zonos-auth-graph` like in `documentation`?

**Review by goodpickle — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Thienhuynh95) [reply package.json:12]: The reason for this node version on package.json is just for unit testing to use built-in `fetch` of node 18

File: .github/actions/auto-assign/package.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  ],
+  "scripts": {
+    "pack": "ncc -m -o . build lib/index.js",
+    "build": "tsc && yarn run pack"
+  },
+  "devDependencies": {
+    "@octokit/webhooks-types": "6.9.0",
+    "@types/node": "^18.11.0",
+    "@vercel/ncc": "0.34.0",
+    "typescript": "^4.9.4"
```
Comment (evad1n) [inline .github/actions/auto-assign/package.json]: There is a newer version of typescript, what do you think about using the latest for this new package?

Comment (evad1n) [reply .github/actions/auto-assign/package.json]: haha whoops this is the wrong package.json, still valid technically but you can ignore.

File: package.json (hunk 1 above)
Comment (Thienhuynh95) [reply package.json:12]: This's removed in the packed version, so the consuming project will not be limited to the node version. I just restrict it for whoever developing this project. Below is what consuming project will get.
image

File: package.json [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    "rollup": "^3.20.2",
+    "rollup-plugin-progress": "^1.1.2",
+    "rollup-plugin-sizes": "^1.0.5",
+    "rollup-plugin-typescript2": "^0.34.1",
+    "ts-jest": "^29.0.5",
+    "ts-node": "^10.9.1",
+    "ts-toolbelt": "^9.6.0",
+    "tsconfig-paths-webpack-plugin": "^4.0.1",
+    "ttypescript": "^1.5.15",
+    "typescript": "^4.9.5",
```
Comment (evad1n) [inline package.json:57]: Hmmm, there is a newer version of typescript now, but I know from experience that the `ttypescript` doesn't work yet with it. I wonder if we can find another option. Some quick digging: https://github.com/cevek/ttypescript/issues/140#issuecomment-1488575848

File: .github/actions/auto-assign/package.json (hunk 5 above)
Comment (Thienhuynh95) [reply .github/actions/auto-assign/package.json]: It's valid. Upgraded. Thanks for catching that.

File: package.json (hunk 6 above)
Comment (Thienhuynh95) [reply package.json:57]: Yah, at first I upgraded to typescript 5, but I downgraded back to 4.9.5. And that's the reason why 😢 
I'm not sure what will be the alternative yet, but I will do more research on it.

Comment (evad1n) [reply package.json:57]: Even if there isn't anything ready yet. I think it's worth keeping up to date to keep up with newer versions of typescript.

File: codegen/auth.codegen.ts (hunk 4 above)
Comment (Thienhuynh95) [reply codegen/auth.codegen.ts]: Sounds good. Thanks for the suggestion.

File: src/gqlRequest.ts (hunk 3 above)
Comment (Thienhuynh95) [reply src/gqlRequest.ts]: 😆 Good catch. Thanks

File: pull_request_template.md (hunk 2 above)
Comment (Thienhuynh95) [reply pull_request_template.md]: Thanks for the suggestion.

**Review by goodpickle — APPROVED** [review]

File: package.json (hunk 6 above)
Comment (Thienhuynh95) [reply package.json:57]: Yes, I agree, but there isn't a version of `ttypescript` or `ts-patch` that works with typescript 5 yet. I'll try to upgrade when there is one. Thanks

**Review by evad1n — APPROVED** [review]

**Pull request merged**